### DISCUSSION TITLE
ini files are not executed on new session

### DISCUSSION THREAD

**Discussion by abyrnoradan** [WinEdt-Team/WinEdt-Support#64] [1.1 General Help and Discussion]
Hello,
I defined many shortkeys, macros etc in tool.ini ou mainmenu.ini
but when i open winedt all these items are not active. I have to lauch "open interface" then click tool and mainmenu then "load current script".

Why ?
Did I miss something to have theses files run by default ?

**Comment by Alex-Simonic** [comment]
Yes, you are missing something. It appears that you embarked on extensive customizations but neglected to read the introduction to the Configuration Manual (Help). It explains how to use the Options Interface to make and maintain your local customizations.

No ini files are loaded on startup. Nor is there any need to load them when you restart WinEdt. A compiled version of all loaded ini files is used (as it is a few 100 times faster to load compared to parsing ini files). If you want to create you customizations outside the default ini files you should place them in Private.ini or use the  `%include="tool.ini"` etc... in your Private.ini. Otherwise, your custom ini files (other than those available through the Options Interface) will not be loaded when you execute the Rebuild All or Local commands. This probably explains why you believe that WinEdt makes changes to your settings (your use of the Rebuild All is probably the cause - this command seldom needs to be used).

Hope this clarifies a few things...

**Reply by abyrnoradan** [reply]
Yes ! Thanks a lot ! 
I am so used to use "prehistorical" winedt that i need some time to get used to this new one. Everything was writen in Winedt.ini which was loaded each time and "basta". Another time ....

first : Where do i have to save my "perso.ini" ?
and after that
do I just have to "upgrade local"  ?
And everything will work ?

**Reply by abyrnoradan** [reply]
During these past 3 hours I tried to have my personnal shorcuts available when launching winedt.
without the least succes .... I become crazy !

As you said 
I wrote my definitions in private.ini in %B/mainmenu.ini in %b mainmenu.ini  I upgrade I rebuilt local, global, all,  etc etc etc 
Nothing !
And i didn't find in the help the least hint  "how to save your shortcut?" I read, read, read,  nothing ! not a word !
I didn't find any hint "how to re-compile" a new "version of all loaded the ini files" as you said. An analogue od "rebuilt database" in miktex.

Can you explain me clearly what i have to do to have this
ITEM="$$"
    CAPTION="$$"
    MACRO="[Ins('$$');CharLeft()]"
    SHORTCUT="112::F1"

available when i open winedt ?

**Comment by abyrnoradan** [comment]
During these past 3 hours I tried to have my personnal shorcuts available when launching winedt.
without the least succes .... I become crazy !

As you said
I wrote my definitions in private.ini in %B/mainmenu.ini in %b mainmenu.ini I upgrade I rebuilt local, global, all, etc etc etc
Nothing !
And i didn't find in the help the least hint "how to save your shortcut?" I read, read, read, nothing ! not a word !
I didn't find any hint "how to re-compile" a new "version of all loaded the ini files" as you said. An analogue od "rebuilt database" in miktex.

Can you explain me clearly what i have to do to have this
ITEM="$$"
CAPTION="$$"
MACRO="[Ins('$$');CharLeft()]"
SHORTCUT="112::F1"

available when i open winedt ?

**Reply by Alex-Simonic** [reply]
In the Options Menu click on the Options Interface. In this interface (on the right side of WinEdt Window) click on the Help button (top left) and read the instructions on how to use this functionality. Until then there is no point in explaining anything.

I have no idea where you found the instructions on creating your custom .ini files inside %B and %b. This is complete nonsense and you should never modify anything inside %B...

**Comment by abyrnoradan** [comment]
Thank for answering
i have to precise that i am doing a "cosmological" jump from 5.8 to 11... a prety hard one 
I just tried everything i knew to make my customization effective.
I read how to use this "options interface" and it works, indeed,  but i just simply want my personnal macros, shortkeys, toolbar, button etc to be effective when i open winedt !
I found really curious instead of open winedt and begin to work tobe obliged to open a Perso.ini file to open a Options Interface and so on and click here and there then close all these accessories before working !
Especially if i am the only user of winedt as supposed by the licence !

Some novelty and especially the back and forth pdf ability are realy fantastic and i was waiting for them since many years but this curious management of customization are really annoying and extremly disappointing

If there is no solution i will go back to winedt10 (i hope in this version customization to be more direct as in older one).

**Reply by Alex-Simonic** [reply]
Fo0llow up with https://github.com/WinEdt-Team/WinEdt-Support/discussions/67. It appears that your problem has to do with WinEdt not being able to save its settings and file list to its local folder. For the rest of us settings from the previous session are preserved when WinEdt is restarted. Do you get any warning messages about saving WinEdt.dnt when you close the program?

**Comment by Alex-Simonic** [comment]
This case is now closed. You cannot help those who refuse to be helped. There is no problem with loading ini files. It is all in your head:-) Of course, it would be expected from a user to spend an hour or so to get WinEdt back to their preferences. Just kidding. If there is a twelve year kid or younger ask them to install the program for you. And you will be surprised how well things work. No quantum leaps are required.
